Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM python:3.12.0a7-buster as base
+
+# Install poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH /root/.local/bin:$PATH
+
+# Make workspace
+WORKDIR /app
+COPY poetry.toml .
+COPY pyproject.toml .
+
+# Install poetry dependencies
+RUN poetry install
+
+# Copy production files
+FROM base as production
+
+COPY todo_app/templates/layout.html todo_app/templates/
+COPY todo_app/templates/prod_index.html todo_app/templates/index.html
+COPY todo_app/*.py todo_app/
+COPY todo_app/data/*.py todo_app/data/
+
+EXPOSE 8000
+CMD poetry run gunicorn --bind 0.0.0.0 "todo_app.app:create_app()"
+
+FROM base as development
+
+EXPOSE 5000
+CMD poetry run flask run --host 0.0.0.0
+
+FROM base as testing
+
+COPY tests/conftest.py tests/
+COPY tests/integration/*.py tests/integration/
+COPY tests/unit/*.py tests/unit/
```

Comment:
Similarly to above, I'd be tempted to move this to a broader copy of the entire `tests` directory - every sub-folder of tests is going to need a change to Dockerfile, and it would be _really_ easy for someone to forget, and then your testing process isn't actually running a portion of your tests!

Reply:
I might have accidentally done exactly this and forgot to COPY a new file while implementing other review changes ...